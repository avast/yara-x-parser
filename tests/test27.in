//Rule comment
rule test
{
	//Rule block comment
	strings:
		$a = "foo"
		$b = "bar"
	condition:
		$a at 100 and $b at 200
}
