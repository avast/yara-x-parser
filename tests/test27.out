SOURCE_FILE@0..125
  RULE@0..124
    COMMENT@0..14 "//Rule comment"
    WHITESPACE@14..15 "\n"
    RULE_KW@15..19 "rule"
    WHITESPACE@19..20 " "
    IDENTIFIER@20..24 "test"
    WHITESPACE@24..25 "\n"
    BLOCK_EXPR@25..124
      L_BRACE@25..26 "{"
      WHITESPACE@26..28 "\n\t"
      STRINGS@28..84
        COMMENT@28..48 "//Rule block comment"
        WHITESPACE@48..50 "\n\t"
        STRINGS_KW@50..57 "strings"
        COLON@57..58 ":"
        WHITESPACE@58..61 "\n\t\t"
        VARIABLE_STMT@61..71
          VARIABLE@61..63 "$a"
          WHITESPACE@63..64 " "
          ASSIGN@64..65 "="
          WHITESPACE@65..66 " "
          PATTERN@66..71
            STRING_LIT@66..71 "\"foo\""
        WHITESPACE@71..74 "\n\t\t"
        VARIABLE_STMT@74..84
          VARIABLE@74..76 "$b"
          WHITESPACE@76..77 " "
          ASSIGN@77..78 "="
          WHITESPACE@78..79 " "
          PATTERN@79..84
            STRING_LIT@79..84 "\"bar\""
      WHITESPACE@84..86 "\n\t"
      CONDITION@86..122
        CONDITION_KW@86..95 "condition"
        COLON@95..96 ":"
        WHITESPACE@96..99 "\n\t\t"
        EXPRESSION_STMT@99..122
          BOOLEAN_EXPR@99..122
            BOOLEAN_TERM@99..108
              VARIABLE@99..101 "$a"
              WHITESPACE@101..102 " "
              VARIABLE_ANCHOR@102..108
                AT_KW@102..104 "at"
                WHITESPACE@104..105 " "
                PRIMARY_EXPR@105..108
                  INT_LIT@105..108 "100"
            WHITESPACE@108..109 " "
            AND_KW@109..112 "and"
            WHITESPACE@112..113 " "
            BOOLEAN_EXPR@113..122
              BOOLEAN_TERM@113..122
                VARIABLE@113..115 "$b"
                WHITESPACE@115..116 " "
                VARIABLE_ANCHOR@116..122
                  AT_KW@116..118 "at"
                  WHITESPACE@118..119 " "
                  PRIMARY_EXPR@119..122
                    INT_LIT@119..122 "200"
      WHITESPACE@122..123 "\n"
      R_BRACE@123..124 "}"
  WHITESPACE@124..125 "\n"
